src/dispensary.js

```diff
@@ -65,13 +67,36 @@ export default class Dispensary {
       });
   }
 
-  getLibraries() {
+  // Matches only against cached hashes; this is the API external apps and
+  // libraries would use.
+  match(contents, _hashesFile=DEFAULT_HAHES_FILE, _cachedHashes=null) {
+    if (this._cachedHashes === null) {
+      if (_cachedHashes !== null) {
+        this._cachedHashes = _cachedHashes;
+      } else {
+        this._cachedHashes = this._getCachedHashes(_hashesFile);
+      }
+    }
+
+    for (let hashEntry of this._cachedHashes) {
+      let hash = hashEntry.split(' ')[0];
+      let library = hashEntry.split(' ')[1];
+
+      if (hasher(contents) === hash) {
```

Comment:
Yeah good point. I think it used to be a whole class but now it isn't. I'll change it to something more function-like.

- tofumatt (Sent from mobile)

> On 19 Jan 2016, at 19:32, Mike Nolan <notifications@github.com> wrote:
> 
> In src/dispensary.js:
> 
> > +  // Matches only against cached hashes; this is the API external apps and
> > +  // libraries would use.
> > +  match(contents, _hashesFile=DEFAULT_HAHES_FILE, _cachedHashes=null) {
> > +    if (this._cachedHashes === null) {
> > +      if (_cachedHashes !== null) {
> > +        this._cachedHashes = _cachedHashes;
> > +      } else {
> > +        this._cachedHashes = this._getCachedHashes(_hashesFile);
> > +      }
> > +    }
> > +
> > +    for (let hashEntry of this._cachedHashes) {
> > +      let hash = hashEntry.split(' ')[0];
> > +      let library = hashEntry.split(' ')[1];
> > +
> > +      if (hasher(contents) === hash) {
> Does hasher() create a hash? Perhaps createHash or just hash might make more sense for this function name (should have caught this earlier)
> 
> —
> Reply to this email directly or view it on GitHub.
>